EIPS/eip-4804.md

```diff
@@ -0,0 +1,160 @@
+---
+eip: 4804
+title: Web3 URL to EVM Call Message Translation
+description: A translation of an HTTP-style Web3 URL to an EVM call message
+author: Qi Zhou (@qizhou), Chao Pi (@pichaoqkc), Sam Wilson (@SamWilsn)
+discussions-to: https://ethereum-magicians.org/t/eip-4804-web3-url-to-evm-call-message-translation/8300
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-02-14
+---
+
+## Abstract
+
+This standard translates an RFC 2396 URI like `web3://uniswap.eth/` to an EVM message such as:
```

Comment:
I'm not sure I understand what you mean by "EVM message" ? This isn't very common terminology - I think it would be better to say something more along the lines of "it's a standard way to represent an `eth_call`" (h/t @SamWilsn ).

Reply:
Perhaps `EVM message` is a bit old term as I borrowed from pyethereum.  It represents a call to an address with data such as `calldata`, `gasPrice`, etc.   The following is the code snippet copied from pyethereum repo.

![image](https://user-images.githubusercontent.com/2541286/166499989-f48cad6a-9cb7-4859-9076-b6d6f3d63981.png)

https://github.com/ethereum/pyethereum/blob/2f59abb10715049196aeef062931f65495a6bf8d/ethereum/fastvm.py#L63

Meanwhile, looks like in `geth`, it does have explicit `Message` term and just call a `Contract` instead. https://github.com/ethereum/go-ethereum/blob/53304ff6c7b6afdff0f6c338e5480b8112080e2c/core/vm/contract.go#L45

I am open to see what is the better term to describe here.

Reply:
Just re-examine Geth and found the Call Message is defined as follows.

https://github.com/ethereum/go-ethereum/blob/0169d579d0eed4f6366697985a7b0f0b99402783/interfaces.go#L134